src/hapi-waterline.js

```diff
@@ -7,56 +7,79 @@ var requireDir = require('require-dir'),
     _ = require('lodash'),
     Waterline = require('waterline'),
     orm = new Waterline();
+
 // just for testing
 exports.reset = function () {
     orm = new Waterline();
 };
 
-exports.register = function (server, options, next) {
-    var adapters = options.adapters || {},
-        connections = options.connections || {},
-        modelsDefault = options.models,
-        bindFlag = options.decorateServer || false;
+
+
+
+/**
+ *
+ *
+ *
+ * @type {{register: exports.plugin.register, pkg}}
+ */
+exports.plugin = {
+    register: async function (server, options) {
+        var adapters = options.adapters || {},
+            datastores = options.datastores || {},
+            modelsDefault = options.models,
+            bindFlag = options.decorateServer || false;
         path = options.path || [];
-    if (bindFlag) {
-        server.decorate('server', 'getModel', function (model) {
-            return server.plugins['hapi-waterline'].models[model];
+        if (bindFlag) {
+            server.decorate('server', 'getModel', function (model) {
+                return server.plugins['hapi-waterline'].models[model];
+            });
+        }
+
+        if (_.isString(path)) {
+            path = [path];
+        }
+
+        _(path).forEach(function (item, index, collection) {
+            var models = requireDir(item, {recurse: true});
+
+            var extendedModels = _(models).map(function (model, key, object) {
+
+                if (modelsDefault) {
+                    _(modelsDefault).forEach(function (value, key, object) {
+                        if (typeof (model[key]) === 'undefined') {
+                            model[key] = value;
+                        }
+                    });
+                }
+
+                return Waterline.Collection.extend(model);
+            });
+            _(extendedModels).forEach(function (extendedModel) {
+                orm.registerModel(extendedModel)
+            });
         });
-    }
-    if (_.isString(path)) {
-        path = [path];
-    }
-    _(path).forEach(function (item, index, collection) {
-        var models = requireDir(item, {recurse: true});
-        var extendedModels = _(models).map(function (model, key, object) {
-            if (modelsDefault) {
-                _(modelsDefault).forEach(function (value, key, object) {
-                    if (typeof (model[key]) === 'undefined') {
-                        model[key] = value;
-                    }
-                });
+
+
+        await new Promise((resolve, reject) => {
```

Comment:
It looks like you applied ES5 instead of ES2017, but await and async was missed :)

Reply:
Any way good update. I will check it on fresh mind and planning to merge to dev and make local check by your version of README.md

Reply:
Should not be there any more

Reply:
Sorry did not remove it.. Changed it now. However what is you take on using await and async? I used it since it is used in hapi 17.